presto-jmx/src/main/java/com/facebook/presto/connector/jmx/JmxRecordSetProvider.java

```diff
@@ -62,116 +67,154 @@ public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorS
         requireNonNull(columns, "columns is null");
         checkArgument(!columns.isEmpty(), "must provide at least one column");
 
+        ImmutableMap<String, Type> columnTypes = getColumnTypes(columns);
```

Comment:
This code relays (and was relaying before my changes) on the fact that it is ImmutableMap. Please check this comment:
https://github.com/prestodb/presto/pull/4440/files/315e8aec29737e1352dfbf78651ebfd38a0019ed#r59444582